addons/mail/models/mail_message.py

```diff
@@ -1007,11 +1007,22 @@ def _message_format(self, fnames):
                     })
 
             if message_sudo.model and message_sudo.res_id:
-                record_name = self.env[message_sudo.model] \
+                record = self.env[message_sudo.model] \
                     .browse(message_sudo.res_id) \
                     .sudo() \
-                    .with_prefetch(thread_ids_by_model_name[message_sudo.model]) \
-                    .display_name
+                    .with_prefetch(thread_ids_by_model_name[message_sudo.model])
+                record_name = record.display_name
+
+                if fetch_followers:
```

Comment:
I'm actually concerned by the security aspect of this coming from manually allowing the user to set this param on the RPC. Are we sure we properly check access right to record before using sudo in all cases?